### DISCUSSION TITLE
No data in indices

### DISCUSSION THREAD

**Discussion by jeffprandall** [pfelk/pfelk#351] [Q&A]
I feel like a noob here.  Fresh ubuntu 20 install, did all the steps, everything seems up and running, charts and templates all imported, OPNsense updated and forwarding the logs.  Still no data.  When I run _ss -aunp_ I see something is listening on ports 5140 and 5141 but if run _nmap -sU -p 5140 192.168.2.15_ from a remote computer it shows the port is closed.  UFW is Inactive.

**Comment by a3ilson** [comment]
have you tried running tcp dump to confirm logs are being received/sent?

**Comment by jeffprandall** [comment]
Here are the results

`jeff@dh4:~$ sudo tcpdump -i ens18 -nn -s0 -v port 5140
[sudo] password for jeff:
tcpdump: listening on ens18, link-type EN10MB (Ethernet), capture size 262144 bytes
23:52:27.931797 IP (tos 0x0, ttl 64, id 42214, offset 0, flags [none], proto UDP (17), length 240)
    192.168.2.1.514 > 192.168.2.15.5140: SYSLOG, length: 212
        Facility auth (4), Severity alert (1)
        Msg: Nov  6 16:52:27 snort[66208]: [120:18:3] (http_inspect) PROTOCOL-OTHER HTTP server response before client request  [Classification: Unknown Traffic] [Priority: 3] {TCP} 104.154.126.176:80 -> 24.23.47.99:52873
23:52:27.931998 IP (tos 0x0, ttl 64, id 50840, offset 0, flags [none], proto UDP (17), length 239)
    192.168.2.1.514 > 192.168.2.15.5140: SYSLOG, length: 211
        Facility auth (4), Severity alert (1)
        Msg: Nov  6 16:52:27 snort[66208]: [120:18:3] (http_inspect) PROTOCOL-OTHER HTTP server response before client request  [Classification: Unknown Traffic] [Priority: 3] {TCP} 104.154.126.176:80 -> 24.23.47.99:2058
23:52:28.415560 IP (tos 0x0, ttl 64, id 47602, offset 0, flags [none], proto UDP (17), length 129)
    192.168.2.1.514 > 192.168.2.15.5140: SYSLOG, length: 101
        Facility kernel (0), Severity info (6)
        Msg: Nov  6 16:52:28 kernel: arp: 192.168.2.8 moved from 00:25:90:a7:68:3d to 00:25:90:a7:68:3c on bce1
23:52:29.931946 IP (tos 0x0, ttl 64, id 17896, offset 0, flags [none], proto UDP (17), length 239)
    192.168.2.1.514 > 192.168.2.15.5140: SYSLOG, length: 211
        Facility auth (4), Severity alert (1)
        Msg: Nov  6 16:52:29 snort[66208]: [120:18:3] (http_inspect) PROTOCOL-OTHER HTTP server response before client request  [Classification: Unknown Traffic] [Priority: 3] {TCP} 35.190.244.186:80 -> 24.23.47.99:31232
23:52:30.934360 IP (tos 0x0, ttl 64, id 42259, offset 0, flags [none], proto UDP (17), length 239)
    192.168.2.1.514 > 192.168.2.15.5140: SYSLOG, length: 211
        Facility auth (4), Severity alert (1)
        Msg: Nov  6 16:52:30 snort[66208]: [120:18:3] (http_inspect) PROTOCOL-OTHER HTTP server response before client request  [Classification: Unknown Traffic] [Priority: 3] {TCP} 104.154.126.176:80 -> 24.23.47.99:2058
23:52:31.673765 IP (tos 0x0, ttl 64, id 30680, offset 0, flags [none], proto UDP (17), length 195)
    192.168.2.1.514 > 192.168.2.15.5140: SYSLOG, length: 167
        Facility local0 (16), Severity info (6)
        Msg: Nov  6 16:52:31 filterlog[51554]: 4,,,1000000103,bce0,match,block,in,4,0x20,,223,23649,0,none,6,tcp,40,45.134.26.237,24.23.47.99,42775,44128,0,S,238673762,,1024,,
23:52:31.673765 IP (tos 0x0, ttl 64, id 53853, offset 0, flags [none], proto UDP (17), length 196)
    192.168.2.1.514 > 192.168.2.15.5140: SYSLOG, length: 168
        Facility local0 (16), Severity info (6)
        Msg: Nov  6 16:52:31 filterlog[51554]: 4,,,1000000103,bce0,match,block,in,4,0x20,,223,22356,0,none,6,tcp,40,45.135.232.119,24.23.47.99,57605,9138,0,S,3021101957,,1024,,
23:52:32.673669 IP (tos 0x0, ttl 64, id 58876, offset 0, flags [none], proto UDP (17), length 215)
    192.168.2.1.514 > 192.168.2.15.5140: SYSLOG, length: 187
        Facility local0 (16), Severity info (6)
        Msg: Nov  6 16:52:32 filterlog[51554]: 91,,,1592882599,bce0,match,pass,in,4,0x20,,37,9008,0,DF,6,tcp,60,45.144.225.175,192.168.2.9,52456,22,0,S,3458893192,,29200,,mss;sackOK;TS;nop;wscale
23:52:36.673658 IP (tos 0x0, ttl 64, id 30347, offset 0, flags [none], proto UDP (17), length 175)
    192.168.2.1.514 > 192.168.2.15.5140: SYSLOG, length: 147
        Facility local0 (16), Severity info (6)
        Msg: Nov  6 16:52:36 filterlog[51554]: 6,,,1000000105,bce1,match,block,in,6,0x00,0x00000,64,UDP,17,36,fe80::225:90ff:fea7:6453,ff02::1:2,546,547,36
23:52:37.673706 IP (tos 0x0, ttl 64, id 29587, offset 0, flags [none], proto UDP (17), length 192)
    192.168.2.1.514 > 192.168.2.15.5140: SYSLOG, length: 164
        Facility local0 (16), Severity info (6)
        Msg: Nov  6 16:52:37 filterlog[51554]: 4,,,1000000103,bce0,match,block,in,4,0x20,,231,8082,0,none,6,tcp,40,94.232.45.4,24.23.47.99,55250,22980,0,S,724855093,,1024,,
23:52:37.934716 IP (tos 0x0, ttl 64, id 48201, offset 0, flags [none], proto UDP (17), length 240)
    192.168.2.1.514 > 192.168.2.15.5140: SYSLOG, length: 212
        Facility auth (4), Severity alert (1)
        Msg: Nov  6 16:52:37 snort[66208]: [120:18:3] (http_inspect) PROTOCOL-OTHER HTTP server response before client request  [Classification: Unknown Traffic] [Priority: 3] {TCP} 104.154.126.176:80 -> 24.23.47.99:52873`

And in PfSense in Status > System Logs > General it says: Nov 6 16:41:18 | syslogd |   | sendto: Host is down

**Comment by a3ilson** [comment]
@jeffprandall - based on the output provided above, you are sending pfSense logs via TCP. The base setup is configured for UDP.

You can adjust pfSense to send via UDP 
or
You can amend 01-inputs.con to listen on tcp vs udp

**Reply by jeffprandall** [reply]
I have the standard remote logging enabled so I don't know how its sending TCP.
![image](https://user-images.githubusercontent.com/119567/140632497-649d6d26-76a6-4734-94ad-57915d47e545.png)

**Comment by a3ilson** [comment]
@jeffprandall - I am not running pfSense and unsure where that setting would reside. However @revere521 might be able to step some insight on the topic.  Additionally, to get you up and running, simply revise the 01-inputs.conf to listen on tcp vs udp, save and restart Logstash. 

Revised 01-inputs.conf below, once updated restart Logstash and you should be good-to-go.
```
# 01-inputs.conf
################################################################################
# Version: 21.02                                                               #
# Required: Yes                                                                #
# Description: Sets the type and port to listen. User may remove unutilized    #
# inputs, as desired.                                                          #
################################################################################
#
input {
  ### Firewall 1 ###
  tcp {
    id => "pfelk-1"
    type => "firewall-1"
    port => 5140
    #ssl => true
    #ssl_certificate_authorities => ["/etc/logstash/ssl/YOURCAHERE.crt"]
    #ssl_certificate => "/etc/logstash/ssl/SERVER.crt"
    #ssl_key => "/etc/logstash/ssl/SERVER.key"
    #ssl_verify_mode => "force_peer"
  }
  ### Firewall 2 ###
  udp {
    id => "pfelk-2" 
    type => "firewall-2"
    port => 5141
    #ssl => true
    #ssl_certificate_authorities => ["/etc/logstash/ssl/YOURCAHERE.crt"]
    #ssl_certificate => "/etc/logstash/ssl/SERVER.crt"
    #ssl_key => "/etc/logstash/ssl/SERVER.key"
    #ssl_verify_mode => "force_peer"
  }
  ### Suricata ###
  tcp {
    id => "pfelk-suricata"
    type => "suricata"
    port => 5040
    #ssl => true
    #ssl_certificate_authorities => ["/etc/logstash/ssl/YOURCAHERE.crt"]
    #ssl_certificate => "/etc/logstash/ssl/SERVER.crt"
    #ssl_key => "/etc/logstash/ssl/SERVER.key"
    #ssl_verify_mode => "force_peer"
  }
  ### HAProxy ###
  udp {
    id => "pfelk-haproxy"
    type => "haproxy"
    port => 5190
    #ssl => true
    #ssl_certificate_authorities => ["/etc/logstash/ssl/YOURCAHERE.crt"]
    #ssl_certificate => "/etc/logstash/ssl/SERVER.crt"
    #ssl_key => "/etc/logstash/ssl/SERVER.key"
    #ssl_verify_mode => "force_peer"
  }
  ### Beats ###
  beats {
    id => "Beats"
    type => "beats"
    port => 5044
  }
}
```

**Reply by jeffprandall** [reply]
I tried that and it also didn't work.  I don't know if it makes a difference but I am running the Docker version of pfelk.  In the docker-compose I did switch 5140 to tcp.  Something interesting is I on the docker host I see its listening on port 5140
![image](https://user-images.githubusercontent.com/119567/140659580-075aae9d-c3c3-4b78-b4ad-be4bf0675eda.png)

But when I try and see if its open remotely I get thats its closed.
![image](https://user-images.githubusercontent.com/119567/140659612-1388e521-2fe9-45ae-9aed-95e27bc561d2.png)
